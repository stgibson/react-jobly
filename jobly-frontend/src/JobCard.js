import React from "react";
import Card from "react-bootstrap/Card";

/**
 * Component for rendering info of a job in a card
 * @param
 * {Object{Object{string|boolean|Object{number}}}|Object{number|string}|function}
 * param0 
 * @returns JSX code for rendering a job card
 */
const JobCard = ({ user, job, apply }) => {
  return (
    <Card data-testid={ job.id }>
      <Card.Body>
        <Card.Title>
          { job.title }
        </Card.Title>
        <Card.Text>
          { `Salary: ${job.salary ? job.salary : ""}` }
        </Card.Text>
        <Card.Text>
          { `Equity: ${job.equity ? job.equity : ""}` }
        </Card.Text>  
      </Card.Body>
    </Card>
  );
};

export default JobCard;