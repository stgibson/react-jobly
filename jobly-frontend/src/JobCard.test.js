import React from "react";
import { MemoryRouter } from "react-router-dom";
import { render } from "@testing-library/react";
import JobCard from "./JobCard";

const job = {
  id: 1,
  title: "Job",
  salary: 100,
  equity: "0.5"
};

const findAllCompanies = jest.fn(() => null); // Learned how to mock functions at https://jestjs.io/docs/mock-functions

it("renders without crashing", () => {
  render(<MemoryRouter><JobCard job={ job } /></MemoryRouter>);
});

it("matches snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter><JobCard job={ job } /></MemoryRouter>
  );
  
  expect(asFragment()).toMatchSnapshot();
});

it("shows job details", () => {
  const { queryByText } = render(
    <MemoryRouter><JobCard job={ job } /></MemoryRouter>
  );

  expect(queryByText(job.title)).toBeInTheDocument();
  expect(queryByText(`Salary: ${job.salary}`)).toBeInTheDocument();
  expect(queryByText(`Equity: ${job.equity}`)).toBeInTheDocument();
});