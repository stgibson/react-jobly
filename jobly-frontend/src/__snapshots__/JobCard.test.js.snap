// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="card"
    data-testid="1"
  >
    <div
      class="card-body"
    >
      <div
        class="card-title h5"
      >
        Job
      </div>
      <p
        class="card-text"
      >
        Salary: 100
      </p>
      <p
        class="card-text"
      >
        Equity: 0.5
      </p>
    </div>
  </div>
</DocumentFragment>
`;
